modules/internal/ChapelLocale.chpl

```diff
@@ -247,6 +247,35 @@ module ChapelLocale {
       return "";
     }
 
+    //
+    // Support for different types of memory:
+    // large, random-access (low latency), and streaming (high bandwidth)
+    // Inspired by madvise()'s support for RANDOM and SEQUENTIAL access
+    //
+    pragma "no doc"
+    proc defaultMemory() : locale {
+      _throwPVFCError();
+      return this;
+    }
+
+    pragma "no doc"
+    proc largeMemory() : locale {
+      _throwPVFCError();
+      return this;
+    }
+
+    pragma "no doc"
+    proc randomMemory() : locale {
+      _throwPVFCError();
+      return this;
+    }
+
+    pragma "no doc"
+    proc streamingMemory() : locale {
+      _throwPVFCError();
+      return this;
+    }
+
```

Comment:
If nobody objects, I will change randomMemory to lowLatencyMemory, and streamingMemory to highBandwidthMemory.  I won't push that change right away because it will hide this part of the github conversation.